java-checks/src/test/files/checks/FileCreateTempFileCheck.java

```diff
@@ -0,0 +1,41 @@
+package foo;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+class A {
+  static File b = File.createTempFile("", "");
+  boolean mkdir = b.mkdir();
+
+  static {
+    b = File.createTempFile("", "");
+  }
+
+  private static File b() {
+    return b;
+  }
+
+  private void noncompliant() throws IOException {
```

Comment:
In fact, AFAIUnderstand, you can detect cases way more complex than this one where calls don't have to be contiguous. For sake of documentation there should be case like this in tests.